src/charm.py

```diff
@@ -232,14 +245,25 @@ def _reconcile(self) -> None:
                     self._ingress_provider
                 )
                 tls_information = TLSInformation.from_charm(self, self.certificates)
+                self.unit.set_ports(80, 443)
                 self.haproxy_service.reconcile_ingress(
                     config, ingress_requirers_information, tls_information.external_hostname
                 )
             case ProxyMode.LEGACY:
+                required_ports: set[Port] = set()
+                for service in self.reverseproxy_requirer.get_services_definition().values():
+                    port = service["service_port"]
+                    if not validate_port(port):
+                        logger.error("Requested port: %s is not a valid tcp port. Skipping", port)
```

Comment:
If it is logged as an error, shouldn't the check go into blocked state?

Reply:
In terms of the logic, we are continuing execution which indicates that it is not an error for a port to be invalid. It would be worthwhile to think about whether this is an error scenario, what the user would be expecting and so on

Reply:
I implemented it like this because I was thinking about cases where haproxy can be related to multiple backend applications, and I don't think we should go into error state because one of the requirers has an invalid configuration, rather we can inform the operator via log messages.

Also it's not really an error if we're not opening the ports, as it doesn't affect the charm and I think a backend requesting an invalid port will not work regardless of the ingress implementation. I used `logger.error` here just to make the message visible when the operator looks at the logs.

Reply:
Hmm, I think the charm should still go into blocked state. The workload should still serve for all the backends it can, it should be visible to the operator that there is something wrong. The operator is unlikely to go into the logs unless they know to take a look. What do you think?